src/api/controllers/channels.ts

```diff
@@ -64,3 +66,23 @@ export async function handleInternalChannelUpdate(data: DiscordPayload) {
 
   eventHandlers.channelUpdate?.(channel, cachedChannel);
 }
+
+export async function handleInternalInviteCreate(data: DiscordPayload) {
+  if (data.t !== "INVITE_CREATE") return;
+
+  const payload = data.d as InviteCreateEvent;
+  const channel = await cacheHandlers.get("channels", payload.channnel_id);
+  if (!channel) return;
```

Comment:
You should just pass the `InvitePayload` and not retrieve and pass the channel structure from the cache.
```ts
eventHandlers.inviteCreate?.(payload);
```